Packet++/src/UdpLayer.cpp

```diff
@@ -141,6 +142,8 @@ namespace pcpp
 			if (!m_NextLayer)
 				m_NextLayer = new PayloadLayer(udpData, udpDataLen, this, m_Packet);
 		}
+		else if (InfiniBandLayer::isInfiniBandPort(portDst))
```

Comment:
I'd be also useful to implement a static method `InfiniBandLayer::isDataValid(udpData, udpDataLen)` that gets the raw buffer and checks if this might be a valid infiniband packet. Mostly it should probably check that `udpData != nullptr` and that `udpDataLen` is equal or larger than the size of `rxe_bth`

Reply:
okey.